roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
I've tested it a few different ways, and `ansible_connection` is defined and set to `local` when I (or Packer) pass `--connection=local`. Ansible 1.9.3.